File: tests/src/main/webapp/WEB-INF/scripts/rest-tests.scala
import javax.ws.rs.core.MediaType._
import javax.ws.rs.{GET, Path, Produces}



@Path("/scala")
class ScalaRes {

    @GET
    @Produces(Array(TEXT_PLAIN))
    @Path("helloworld")
    def helloworld(): String = {
        return "Hello to Script World for Scala!!!" + new java.util.Date();
    }
}

classOf[ScalaRes]

// /class ResGen extends IResourceGenerator {
//    def apply(app: Application): java.util.Set[Resource] = {
//        return new java.util.HashSet[Resource](
//            Set(Resource.from(classOf[ScalaRes])).asJavaCollection
//        )
//    }
//}
//new ResGen

File: tests/src/main/webapp/WEB-INF/scripts/rest-tests.js
var Resource = Java.type("org.glassfish.jersey.server.model.Resource");
var ResourceMethod = Java.type("org.glassfish.jersey.server.model.ResourceMethod");

var ContainerRequestContext = Java.type("javax.ws.rs.container.ContainerRequestContext");
var MediaType = Java.type("javax.ws.rs.core.MediaType");
var IResourceGenerator = Java.type("org.joints.rest.script.ScriptResLoader.IResourceGenerator");
var Arrays = Java.type(java.util.Arrays.class.getName());
var HashSet = Java.type(java.util.HashSet.class.getName());

load(current_path + "/rest-utils.jsx");

function apply(resourceConfig) { //ScriptResLoader
    var resourceBuilder = Resource.builder();
    resourceBuilder.path("helloworld");

    var methodBuilder = resourceBuilder.addMethod("GET");
    methodBuilder
        .produces(MediaType.TEXT_PLAIN_TYPE)
        .handledBy(
           _endPoint(function(containerRequestContext) {
                    return this.toString() + "\n" + containerRequestContext;
            }));
    var resource = resourceBuilder.build();
    return new HashSet(Arrays.asList(resource));
}
var ResGen = Java.extend(IResourceGenerator, {apply: apply});
new ResGen();